MDM.py

```diff
@@ -72,134 +77,139 @@ def mean_riemann(covmats, tol=10e-9, maxiter=50, init=None,
+                self.__ch,
+                self.__gestures.shape[3],
+            )
+        )
+        for i in range(self.__number_of_gestures):
+            base_new[i, : self.__repetitions, :, :] = self.__gestures[i, :, :, :]
+            base_new[i, self.__repetitions :, :, :] = other.__gestures[i, :, :]
+        return base_new
+
+
 class MDM(object):
-#   Ten Pan dostaje dwie rzeczy - baze stworzoną w sposób powyższy oraz wyciety gest z sygnalu/ow
-#   Zwraca (jeśli będzie szczęśliwy) indeks gestu z bazy który jest najbliższy gestowi live (wegle geometrii riemanna)
-    def __init__(self,baza,gest): #baza - 3d , gest = 3D
-        self.__baza = baza
-        self.__gest = gest
-        self.__ilegestow = baza.shape[0]
-        self.__wyniki = {}
+    #   Accepts base made with class above and 2D matrix of gesture signals
+    #   Returns index of gesture classified based on riemann geometry
+    def __init__(self, base, gesture):  # base - 3d , gesture = 2D
+        self.__Base = base
+        self.__gesture = gesture
+        self.__number_of_gestures = base.shape[0]
+        self.__distances = {}
+
     def classify(self):
-        covGest = np.corrcoef(self.__gest)
-        for gest in range(self.__ilegestow):
-            self.__wyniki[gest] = distance_riemann(self.__baza[gest,:,:],covGest)
-#            print(dist_riem(self.__baza[gest,:,:],covGest))
-#            print(distance_riemann(self.__baza[gest,:,:],covGest))
-        return self.__wyniki
-    def getmin(self , threshold=1):
-#        print(threshold)
-#        print(min(self.__wyniki.values()))
-        if 0>threshold or threshold >1:
-            raise ValueError('Values between 0 and 1 only')
-        elif min(self.__wyniki.values()) < threshold:
-            return min(self.__wyniki, key = self.__wyniki.get)
+        # computes distances between online gesture and each gesture from base
+        corrGesture = np.corrcoef(self.__gesture)
+        for gesture in range(self.__number_of_gestures):
+            self.__distances[gesture] = distance_riemann(
+                self.__Base[gesture, :, :], corrGesture
+            )
+        return self.__distances
+
+    def getmin(self, threshold=1):
+        # chooses minimal distance (if bigger than threshold returns None)
+        if 0 > threshold or threshold > 1:
+            raise ValueError("Values between 0 and 1 only")
+        elif min(self.__distances.values()) < threshold:
+            return min(self.__distances, key=self.__distances.get)
         else:
             return None
+
     @property
     def number_of_gestures(self):
-        return self.__ilegestow
+        return self.__number_of_gestures
+
     def __str__(self):
-        print('nr gestu bazy     odległość do gestu live')
-        for gest , odleglosc in self.__wyniki.items():
-            print(str(gest) + '                 ' +str( odleglosc))
-        return '  '
+        print("number of gesture     distance to gesture online")
```

Comment:
Refactor the print, so it prints names and values in one line, not one after the other.